api/v1alpha1/imageupdater_types.go

```diff
@@ -24,81 +24,243 @@ import (
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
 // ImageUpdaterSpec defines the desired state of ImageUpdater
+// It specifies which applications to target, default update strategies,
+// and a list of images to manage.
 type ImageUpdaterSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 
-	// Foo is an example field of ImageUpdater. Edit imageupdater_types.go to remove/update
-	// Foo string `json:"foo,omitempty"`
+	// Namespace indicates the target namespace of the applications.
+	// This is the namespace where the controller will look for Argo CD Applications
+	// matching the criteria in ApplicationRefs.
+	// +kubebuilder:validation:Required
+	Namespace string `json:"namespace"`
+
+	// --- Defaults/Global settings for all applications matched by this CR ---
+	// These can be overridden at the ApplicationRef or ImageConfig level.
+
+	// UpdateStrategy defines the update strategy to apply.
+	// Examples: "semver", "latest", "digest", "name".
+	// This acts as the default if not overridden at a more specific level.
+	// +optional
+	// +kubebuilder:default:="semver"
+	UpdateStrategy *string `json:"updateStrategy,omitempty"`
+
+	// ForceUpdate specifies whether updates should be forced.
+	// This acts as the default if not overridden.
+	// +optional
+	// +kubebuilder:default:=false
+	ForceUpdate *bool `json:"forceUpdate,omitempty"`
+
+	// AllowTags is a regex pattern for tags to allow.
+	// This acts as the default if not overridden.
+	// +optional
+	AllowTags *string `json:"allowTags,omitempty"`
+
+	// IgnoreTags is a list of tags to ignore.
```

Comment:
From https://argocd-image-updater.readthedocs.io/en/latest/configuration/images/#appendix , ignoreTags is a list of glob-like patterns of tags.

Reply:
I can change the comment to something like: `// IgnoreTags is a list of of glob-like patterns of tags to ignore.`

Reply:
I'm also a bit uncomfortable with the mixed use of string vs *string, bool vs *bool in struct fields. I think the common practice in this project is to use values especially for primitives in struct fields.

Reply:
Right, in argocd types.go values are used for primitives https://github.com/argoproj/argo-cd/blob/release-2.13/pkg/apis/application/v1alpha1/types.go